app/lib/features/settings/pages/email_addresses.dart

```diff
@@ -0,0 +1,244 @@
+  bool passwordVisible = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return AlertDialog(
+      title: const Text(
+        'Add email address',
+      ), // The token-reset path is just the process by which control over that email address is confirmed.
+      content: Form(
+        key: _formKey,
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            Padding(
+              padding: const EdgeInsets.all(5),
+              child: TextFormField(
+                controller: newEmailAddress,
+                // FIXME: should have an email-addres-validator ,
+                decoration: const InputDecoration(hintText: 'Email Address'),
+              ),
+            ),
+          ],
+        ),
+      ),
+      actions: <Widget>[
+        TextButton(
+          onPressed: () => Navigator.pop(context, null),
+          child: const Text('Cancel'),
+        ),
+        TextButton(
+          onPressed: () => onSubmit(context),
+          child: const Text('Submit'),
+        ),
+      ],
+    );
+  }
+
+  void onSubmit(BuildContext context) {
+    if (!_formKey.currentState!.validate()) {
+      customMsgSnackbar(context, 'Email or password seems to be not valid.');
+      return;
+    }
+    Navigator.pop(context, newEmailAddress.text);
+  }
+}
+
+class EmailAddressesPage extends ConsumerWidget {
+  const EmailAddressesPage({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final emailAddresses = ref.watch(emailAddressesProvider);
+    return WithSidebar(
+      sidebar: const SettingsMenu(),
+      child: Scaffold(
+        appBar: AppBar(
+          backgroundColor: const AppBarTheme().backgroundColor,
+          elevation: 0.0,
+          title: const Text('Email Addresses'),
+          centerTitle: true,
+          actions: [
+            IconButton(
+              onPressed: () {
+                ref.invalidate(emailAddressesProvider);
+              },
+              icon: Icon(
+                Atlas.refresh_account_arrows_thin,
+                color: Theme.of(context).colorScheme.neutral5,
+              ),
+            ),
+            IconButton(
+              onPressed: () => addEmailAddress(context, ref),
+              icon: Icon(
+                Atlas.plus_circle_thin,
+                color: Theme.of(context).colorScheme.neutral5,
+              ),
+            ),
+          ],
+        ),
+        body: emailAddresses.when(
+          data: (addresses) => buildAddresses(context, addresses),
+          error: (error, stack) {
+            return const Center(
+              child: Text("Couldn't load all email addresses"),
+            );
+          },
+          loading: () => const Center(
+            child: CircularProgressIndicator(),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget buildAddresses(BuildContext context, EmailAddresses addresses) {
```

Comment:
not very fond of using ```Widget``` methods when returns a children of widget or deep tree, it becomes quite expensive on render thread for rebuilds. Would suggest using private ```Stateless``` or  ```Consumer``` widget in that regard.

Reply:
sure.

Reply:
this was actually from a previous PR that is now not part of this PR anymore as it is already in `main`... If you still want to discuss that, let's address it separately.